### PR TITLE
Update lilyplay.jl for PortAudio architecture.

### PR DESCRIPTION
**Pull request by wherrera10**
Tested only on Windows 10, so please check if it works on OSX and Linux.

### CODE REVIEW

**Comment by wherrera10** [comment]
Fixed my editor hating unlauts.

**Comment by ssfrr** [comment]
Thanks for looking into this! I'll check out the PR shortly.

**Comment by wherrera10** [comment]
There was a lot of static on the audio due to the example's using threads with a non-threaded underlying C callback in PortAudio, so I added a mixer to add all the multitasking threading writes into one summed write.

**Comment by wherrera10** [comment]
win32 build failed due to something in WinRPM? Trying the win32 build above with @async again instead of @spawn.

**Comment by ssfrr** [comment]
hmm, that's strange. I'll ask around among the folks more familiar with WinRPM internals what could cause that issue.

as a side note, it's good to back-tick in-line code - especially with julia macros, which otherwise look like github usernames and end up sending notifications to people who probably aren't interested. In this case it notified the users `@async` and `@Spawn`

2nd side note - please give your commits more descriptive messages, currently they all just say "Update lilyplay.jl", so when I see that you've added a new commit it's not clear what your intent was with the new commit. Not a big deal, it just makes my job a little easier when keeping up-to-date with progress.

**Comment by ssfrr** [comment]
I just restarted the win32 build and it went through this time, so probably some temporary network issue or something. Is this PR ready for review?

**Comment by wherrera10** [comment]
yes, it is ready

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by wherrera10** [comment]
I implemented your suggestions. The only one I don't know how to change properly is the line
    channum = "$(current_task())"[27:end]
which attempts to get something resembling a task id for an @async coroutine. This is not the same as that given by myid() which I believe is the OS process ID, and which is generally 1 for all @async coroutines that are created by @async from the main thread.

Do you know a better way? I followed the source code for @async back and found it terminates in a C library call, and I doubt I can extract a less-"brittle" task ID from that library. Would love to learn better?

File: examples/lilyplay.jl [hunk 1]
```diff
@@ -1,11 +1,69 @@
...
+export WriteMixer, writemixed, play
+
+mutable struct WriteMixer
+    mixbufsize::Int64
+    mixed::Array{Float64,1}
+    channels::Dict{String,Channel}
+    WriteMixer(sz) = new(sz, zeros(Float64, sz), Dict{Int,Channel}())
+end
+
+function writemixed(mixer, writestream)
```
[the remark this answers is not in the document]
Comment (ssfrr) [reply examples/lilyplay.jl]: did you see this note? You can avoid all the task ID and `Channel` stuff with an approach  more like this.

**Comment by wherrera10** [comment]
see updated request of 26 March 2020

**Pull request closed without merging**